src/shared_modules/utils/zlibHelper.hpp

```diff
@@ -88,6 +99,126 @@ namespace Utils
+                    {
+                        throw std::runtime_error {"A potentially insecure path was found: " + outputFilepath.string()};
+                    }
+
+                    const auto isDir {Utils::endsWith(outputFilepath.string(), "/")};
+                    if (isDir)
+                    {
+                        // Create output directory.
+                        std::filesystem::create_directory(outputFilepath);
+                    }
+                    else
+                    {
+                        // Create outputfile.
+                        std::ofstream outFile {outputFilepath, std::ios::binary};
+                        if (!outFile.good())
+                        {
+                            throw std::runtime_error {"Unable to create destination file: " + outputFilepath.string()};
+                        }
+
+                        // Read current file content.
+                        unsigned long bytesRead, totalBytesRead {0};
+                        do
+                        {
+                            // Read compressed data by ZIP_BUF_LEN chunks.
+                            std::vector<char> buffer(ZIP_BUF_LEN);
+                            bytesRead = unzReadCurrentFile(spUnzFile.get(), buffer.data(), buffer.size());
+                            totalBytesRead += bytesRead;
+
+                            // Store current chunk into output file.
+                            outFile.write(buffer.data(), bytesRead);
+                        } while (bytesRead > 0);
+
+                        // Close output file to flush the stream and check for error flags.
+                        outFile.close();
+                        if (!outFile.good())
+                        {
+                            throw std::runtime_error {"Error while writing output file: " + outputFilepath.string()};
+                        }
+
+                        // Check total amount of bytes read.
+                        if (totalBytesRead != fileInfo.uncompressed_size)
+                        {
+                            throw std::runtime_error {"Unable to read content of current file: " +
+                                                      std::string(filename)};
+                        }
+                    }
+
+                    if (!isDir)
+                    {
+                        // Push filename into the output vector.
+                        decompressedFiles.push_back(outputFilepath.string());
+                    }
+                }
+
+                // Go to next file within the .zip file.
+                if (currentFileIndex + 1 < globalInfo.number_entry)
+                {
+                    if (unzGoToNextFile(spUnzFile.get()) != UNZ_OK)
+                    {
+                        throw std::runtime_error {"Unable to get next file of: " + zipFilePath.string()};
+                    }
+                }
```

Comment:
Suggestion: I see `unzGoToNextFile` returns:
```
  return UNZ_OK if there is no problem
  return UNZ_END_OF_LIST_OF_FILE if the actual file was the latest.
```

we could replace the main `for` loop for:
```c++
do{
...
}while(unzGoToNextFile(spUnzFile.get()) == UNZ_OK)
```

Reply:
Done!

I added a test with an empty zip file. 

In this case, the `unzOpen` method fails, not reaching the `do-while` block.

Reply:
FYI, the empty zip file was created like this:
```bash
$ zip empty.zip some_file.txt
$ zip -d empty.zip some_file.txt
```